### PR TITLE
[M10-402] Ogury SDK Thumbnail Integration

### PR DESCRIPTION
**Pull request by carlosalcala-freestar**
**[M10-402](https://freestar.atlassian.net/browse/M10-402)**

## Description, Motivation and Context

**What is the goal of ticket?  What are the acceptance criteria?**

Integrate Ogury SDK - Thumbnail

## How Has This Been Tested? 

Manually tested App Build

## Does this change require updates outside of the code, like updating the documentation, adding a feature flag, or adding fields to a DB table? If so, was this done?

N/A

## Any tickets that this is dependent upon?

N/A

## Should reviewers manually test your changes?

N/A

### Evidence
Screen Shot 2022-08-24 at 01 30 57
Screen Shot 2022-08-24 at 01 31 45
Screen Shot 2022-08-24 at 01 33 25

Screen Shot 2022-08-23 at 19 34 19
Screen Recording 2022-08-17 at 20 13 14
Screen Shot 2022-08-17 at 20 12 54
Screen Shot 2022-08-18 at 21 02 29

https://user-images.githubusercontent.com/104378173/186043183-fe4467f0-5687-446f-8ef7-3c87859feed9.mov

### CODE REVIEW

**Review by kkawai — CHANGES_REQUESTED** [review]
@carlosalcala-freestar Where are the whitelist and blacklist handlers?
https://ogury-ltd.gitbook.io/ios/v/master-1/ad-formats/thumbnail-ad#whitelist-bundles
https://ogury-ltd.gitbook.io/ios/v/master-1/ad-formats/thumbnail-ad#blacklist-view-controllers
There needs to be some exposed methods in the ThumbnailAd (core SDK) and also a way to pass the whitelist and blacklist info to the adapter (in this case Ogury adapter)

**Comment by carlosalcala-freestar** [comment]
> @carlosalcala-freestar Where are the whitelist and blacklist handlers? https://ogury-ltd.gitbook.io/ios/v/master-1/ad-formats/thumbnail-ad#whitelist-bundles https://ogury-ltd.gitbook.io/ios/v/master-1/ad-formats/thumbnail-ad#blacklist-view-controllers There needs to be some exposed methods in the ThumbnailAd (core SDK) and also a way to pass the whitelist and blacklist info to the adapter (in this case Ogury adapter)

@kkawai is now implemented thanks 👍

**Comment by carlosalcala-freestar** [comment]
@kkawai @maddapper please review this again is already with white/black lists implementation

File: FreestarAds-Ogury/Source/objc/OguryFSTRPartnerInitializer.m [hunk 1]
```diff
@@ -30,8 +29,8 @@ - (void)runInitialization {
         && self.partners.count > 0
         && [self.partners.firstObject app_id]) {
 
-        OguryConfigurationBuilder *configurationBuilder = [[OguryConfigurationBuilder alloc] initWithAssetKey:[self.partners.firstObject app_id]];
-        [Ogury startWithConfiguration:[configurationBuilder build]];
+        [OguryChoiceManager.sharedManager setupWithAssetKey:[self.partners.firstObject app_id]];
+        [OguryAds.shared setupWithAssetKey:[self.partners.firstObject app_id]];
```
Comment (maddapper) [inline FreestarAds-Ogury/Source/objc/OguryFSTRPartnerInitializer.m]: @carlosalcala-freestar  do u get deprecating warnings?  if so, please use the latest ogury APIs... thx.

File: FreestarAds-Ogury/Source/objc/OguryThumbnailMediator.m [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    FSTRLog(@"OGURY: loadThumbnailAd");
+    FSTRLog(@"OGURY: adunitId %@", [self.mPartner adunitId]);
+
+    if ([self.mPartner adunitId] == nil) {
+        [self partnerAdLoadFailed:@"adunitId is nil"];
+        return;
+    }
+
+    self.startPoint = CGPointMake(100, 500);
+    self.thumbnail = [[OguryAdsThumbnailAd alloc]initWithAdUnitID:[self.mPartner adunitId]];
```
Comment (maddapper) [inline FreestarAds-Ogury/Source/objc/OguryThumbnailMediator.m]: @carlosalcala-freestar  do u get deprecating warnings?  if so, please use the latest ogury APIs... thx.

**Review by maddapper — DISMISSED** [review]
@carlosalcala-freestar  do u get deprecating warnings?  if so, please use the latest ogury APIs... thx.

**Comment by maddapper** [comment]
@carlosalcala-freestar we need to make sure to use the latest ogury APIs and resolve any deprecation warnings.  thx.

File: FreestarAds-Ogury/Source/objc/OguryFSTRPartnerInitializer.m (hunk 1 above)
Comment (carlosalcala-freestar) [reply FreestarAds-Ogury/Source/objc/OguryFSTRPartnerInitializer.m]: Fixing those @maddapper

File: FreestarAds-Ogury/Source/objc/OguryThumbnailMediator.m (hunk 2 above)
Comment (carlosalcala-freestar) [reply FreestarAds-Ogury/Source/objc/OguryThumbnailMediator.m]: Fixing those @maddapper

**Review by kkawai — APPROVED** [review]

**Pull request merged**